app/api/claim/route.ts

```diff
@@ -49,17 +58,24 @@ export async function GET(request: NextRequest) {
  * @param request 
  * @returns { address, network }
  */
-function validateRequest(request: NextRequest) {
+async function validateRequest(request: NextRequest) {
   const headersList = headers();
   const ip = headersList.get("x-real-ip");
   const address = request.nextUrl.searchParams.get("address");
   const userAgent = headersList.get("user-agent");
   const network = request.nextUrl.searchParams.get("network");
 
-  if (!process.env.FAUCET_ADDRESS || !process.env.FAUCET_SEED) {
+  const { remaining } = await ratelimit.limit(ip ?? '127.0.0.1');
+
+  if (!process.env.FAUCET_ADDRESS || !process.env.FAUCET_SEED || !process.env.KV_REST_API_URL || !process.env.KV_REST_API_TOKEN) {
```

Comment:
Would be nice to convert this into a validation fn on entrypoint.

Reply:
this can be done later i feel